packages/neon_framework/lib/src/login/bloc/login_bloc.dart

```diff
@@ -0,0 +1,65 @@
+import 'package:account_repository/account_repository.dart';
+import 'package:bloc/bloc.dart';
+import 'package:equatable/equatable.dart';
+import 'package:meta/meta.dart';
+
+part 'login_event.dart';
+part 'login_state.dart';
+
+@internal
+class LoginBloc extends Bloc<LoginEvent, LoginState> {
+  LoginBloc({
+    required AccountRepository accountRepository,
+    Uri? serverURL,
+    Credentials? credentials,
+  })  : _accountRepository = accountRepository,
+        super(
+          LoginState.initial(
+            serverURL: credentials?.serverURL ?? serverURL,
+            credentials: credentials,
+          ),
+        ) {
+    on<LoginUseQRCode>(_onLoginUseQRCode);
+    on<LoginUrlEntered>(_onLoginUrlEntered);
+    on<LoginServerChecked>(_onLoginServerChecked);
+    on<LoginCredentialsEntered>(_onLoginCredentialsEntered);
+    on<LoginAccountChecked>(_onLoginAccountChecked);
+  }
+
+  final AccountRepository _accountRepository;
+
+  void _onLoginUseQRCode(LoginUseQRCode event, Emitter<LoginState> emit) {
+    emit(const LoginStateScanQrCode());
+  }
+
+  void _onLoginUrlEntered(LoginUrlEntered event, Emitter<LoginState> emit) {
+    emit(
+      LoginStateCheckServer(
+        serverURL: event.serverURL,
+        credentials: event.credentials,
+      ),
+    );
+  }
+
+  void _onLoginServerChecked(LoginServerChecked event, Emitter<LoginState> emit) {
+    switch (state) {
+      case LoginStateCheckServer(:final credentials) when credentials != null:
+        emit(LoginStateCheckAccount(credentials: credentials));
+
+      case LoginStateCheckServer(:final serverURL):
+        emit(LoginStateFlowV2(serverURL: serverURL));
+
+      default:
+        return;
```

Comment:
Blocs should silently ignore events emmited on a wrong state.

Reply:
I disagree. It should throw an error because it is a programming error and developers need to know about it.

Reply:
No; It isn't an error.
Events are computed asynchronously (because of streams) so we must be relaxed about the state.
There is nothing wrong about adding the event. It just doesn't affect the state.

Reply:
But the possible states are already covered, so anything else should fail.
For example if you now add a new possible state you might forget to update this switch and there would never be an error about it.

Reply:
If you are concerned about missing a future case I can explicitly add the missing ones.
I don't think it matters because in this case we either have a special case (that would need to be handled) or we just silently ignore it.
I also can't imagine how adding a new event that needs to be handled would be missed in a unit test.